src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,288 @@
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to set time interval, Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+        }
+
+        /// <summary>
+        /// A property to get "from" value(start) of time interval for statistics rule.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value>from(start) of time interval.</value>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown while setting this property when Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public DateTime From
+        {
+            get
+            {
+                DateTime from;
+                DateTime to;
+                int ret = Interop.Stc.Rule.GetTimeInterval(_ruleHandle, out from, out to);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get time interval(from value) for rule, Error - " + (StcError)ret);
+                }
+                return from;
+            }
+        }
+
+        /// <summary>
+        /// A property to get "to" value(end) of time interval for statistics rule.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <value>to(end) of time interval.</value>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown while setting this property when Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public DateTime To
+        {
+            get
+            {
+                DateTime from;
+                DateTime to;
+                int ret = Interop.Stc.Rule.GetTimeInterval(_ruleHandle, out from, out to);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to get time interval(to value) for rule, Error - " + (StcError)ret);
+                }
+                return to;
+            }
+        }
+
+        /// <summary>
+        /// To get Interface type for statistics rule.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <returns>Interface type.</returns>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown while setting this property when Stc is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown while setting this value due to an invalid operation.</exception>
+        public NetworkInterface GetInterfaceType()
```

Comment:
Why not property?

Reply:
There is an 'Unknown' enum in NetworkInterface. And you told me to throw an exception in case of Unknown. Now property getters don't throw exceptions usually, so i switched to method.

Reply:
Why `Unknown` value are returing?
If a default value is existed as NetworkInterface.All, We can avoid error case

Reply:
This value is setted by User
So, it can't existed a error case

Reply:
When we create rule object, we use Core API "stc_stats_rule_create", it sets interfaceType to unknown('0') by default. And depending on the filtering options, the user might set it or let it remain to be unknown. So it is possible that C# layer may get "unknown" from CAPI. I think that we should not even throw an exception in such case, because it is perfectly valid to get Unknown as NetworkInterface of some Rule.

Reply:
You can use nullable type to present `unknown` as `default`. So, I recommend to use pure c# code making a StcRule class. 

>How about make as pure C# object?
and consider using nullable type

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types

Reply:
Done. Patch pushed. Please review.